src/fetchers/tarball-fetcher.js

```diff
@@ -275,24 +290,43 @@ export default class TarballFetcher extends BaseFetcher {
               integrityValidateStream.pipe(fs.createWriteStream(tarballCachePath)).on('error', reject);
             }
 
-            integrityValidateStream.pipe(extractorStream).on('error', reject);
+            integrityValidateStream.pipe(extractorStream).on('error', async err => {
+              try {
+                // Cleanup before retrying
+                await this.removeDownloadedFiles();
+                if (this.validateError && this.validateError.code == 'EBADSIZE') {
+                  // Content-Length did not match the body size. Retry the request.
+                  if (!queueForRetry(err.message)) {
+                    reject(err);
+                  }
+                  return;
+                }
+                reject(err);
+              } catch (err2) {
```

Comment:
How about renaming this to something more meaningful:
```suggestion
              } catch (retryError) {
```